pkgs/servers/firebird/default.nix

```
@@ -1,91 +1,98 @@
-
-   I'm only interested in the embedder shared libary for now.
-   So everything isn't tested yet
-
-*/
-
-stdenv.mkDerivation rec {
-  version = "2.5.7.27050-0";
+let base = {
   pname = "firebird";
 
-  # enableParallelBuilding = false; build fails
+  meta = with lib; {
+    description = "SQL relational database management system";
+    homepage = "https://firebirdsql.org/";
+    changelog = "https://github.com/FirebirdSQL/firebird/blob/master/CHANGELOG.md";
+    license = [ "IDPL" "Interbase-1.0" ];
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ marcweber ];
+  };
 
-  # http://tracker.firebirdsql.org/browse/CORE-3246
   preConfigure = ''
     makeFlags="$makeFlags CPU=$NIX_BUILD_CORES"
   '';
 
   configureFlags =
-    [ "--with-serivec-port=${builtins.toString port}"
+    [
       "--with-service-name=${serviceName}"
       "--with-system-editline"
       "--with-fblog=/var/log/firebird"
       "--with-fbconf=/etc/firebird"
       "--with-fbsecure-db=/var/db/firebird/system"
     ]
-    ++ (lib.optional  (icu != null) "--with-system-icu")
     ++ (lib.optional superServer "--enable-superserver");
 
-  src = fetchurl {
-    url = "mirror://sourceforge/firebird/Firebird-${version}.tar.bz2";
-    sha256 = "06hp6bq5irqvm3h03s79qjgcc3jsjpq150y9aq7anklx9v4nhfqa";
-  };
-
+  configurePhase = ''
+    ./autogen.sh $configureFlags --prefix=$out
+  '';
   hardeningDisable = [ "format" ];
 
-  # configurePhase = ''
-  #   sed -i 's@cp /usr/share/automake-.*@@' autogen.sh
-  #   sh autogen.sh $configureFlags --prefix=$out
-  # '';
-  buildInputs = [libedit icu automake autoconf libtool];
+  buildInputs = [ libedit icu automake autoconf libtool ];
 
-  # TODO: Probably this hase to be tidied up..
-  # make install requires beeing. disabling the root checks
-  # dosen't work. Copying the files manually which can be found
-  # in ubuntu -dev -classic, -example packages:
-  # maybe some of those files can be removed again
-  installPhase = "cp -r gen/firebird $out";
+  LD_LIBRARY_PATH = lib.makeLibraryPath [ icu ];
 
-  meta = {
-    description = "SQL relational database management system";
-    homepage = "https://www.firebirdnews.org";
-    license = ["IDPL" "Interbase-1.0"];
-    maintainers = [lib.maintainers.marcweber];
-    platforms = lib.platforms.linux;
-    broken = true;
-  };
+  installPhase = ''
+    mkdir $out
+    cp -r gen/Release/firebird/* $out
+  '';
+
+}; in {
+
+  firebird_2_5 = stdenv.mkDerivation (base // rec {
+    version = "2.5.9";
+    rev = "R${builtins.replaceStrings ["."] ["_"] version}";
+    sha256 = "sha256-YyvlMeBux80OpVhsCv+6IVxKXFRsgdr+1siupMR13JM=";
+
+    src = fetchFromGitHub {
+      owner = "FirebirdSQL";
+      repo = "firebird";
+      inherit sha256 rev;
+    };
+
+    configureFlags = base.configureFlags ++ [ "--with-serivec-port=${builtins.toString port}" ] ++ (lib.optional  (icu != null) "--with-system-icu");
```

```suggestion
    configureFlags = base.configureFlags
      ++ [ "--with-serivec-port=${builtins.toString port}" ]
      ++ lib.optional (icu != null) "--with-system-icu";
```

overwriting packages with null is discuraged and a boolean flag should be used instead.

---

+Clean-up. Done.